### DISCUSSION TITLE
Action and AnyCable in the same Rails app

### DISCUSSION THREAD

**Discussion by did** [anycable/anycable#177] [Q&A] [answered]
👋 Anycable team. First of all, a huge thanks to your awesome work!

Here is why I'm asking for some help!

Context: I've got a Rails application in production (hosted in Heroku) which is being used as an API for mobile devices. It uses ActionCable behind the scene.

Goal: we want to use AnyCable in production and based on our tests, it's just a matter of changing the WS URL in the mobile app to the Anycable-go WS server. 
However, we are aware that some mobile devices won't upgrade their application and so they will hit the old WS URL (ActionCable). 

Question: How do we make our Rails API application work with both ActionCable (legacy) and AnyCable without creating a new Heroku application? 
I took a look at https://docs.anycable.io/rails/getting_started?id=gradually-migrating-from-action-cable but it doesn't mention how to fix the problem with `mount ActionCable.server => '/cable'`. 
Actually, I can connect to this endpoint BUT since the adapter is Anycable, the ActionCable server methods are not longer implemented, so no subscriptions, nothing (and NotImplementedError exception in the server logs).
Is there a work-around?

Thanks! 🙏

**Comment by did** [comment, marked as answer]
Alright, so I've re-read very carefully the https://docs.anycable.io/rails/getting_started?id=gradually-migrating-from-action-cable URL. 
It looks like I just have to keep the adapter to **redis** in `config/cable.yml` file and that's it. 
And the Rails API "gRPC" app must use the **anycable** adapter if I'm right.
Is that correct?

**Reply by did** [reply]
It worked 🎉

**Reply by palkan** [reply]
Glad to hear you solved it.

Yeah, to keep Action Cable working along AnyCable, you should use the legacy adapter. 

> And the Rails API "gRPC" app must use the anycable adapter if I'm right.

Yes, in RPC you can switch to `anycable` (and in other non-web processes, e.g., `worker`), but in the `web` process running Action Cable you also need to extend the Action Cable adapter (to send broadcasts to both Action Cable and AnyCable): 

```ruby
# config/initializers/anycable.rb
AnyCable::Rails.extend_adapter!(ActionCable.server.pubsub)
```
